src/hotspot/os/linux/cgroupSubsystem_linux.hpp

```diff
@@ -179,47 +179,69 @@ class CachedMetric : public CHeapObj<mtInternal>{
     }
 };
 
+template <class T>
 class CachingCgroupController : public CHeapObj<mtInternal> {
   private:
-    CgroupController* _controller;
+    T _controller;
```

Comment:
We always expect `T` to be a `T*`, so make it so.

Change:

```c++
T* _controller;
```

and remove the `*` from all instantiations.